lib/cli/interactive-setup/index.js

```diff
@@ -12,12 +12,13 @@ const steps = {
 };
 
 module.exports = async (context) => {
-  context = { ...context, inquirer };
-  for (const step of Object.values(steps)) {
+  context = { ...context, inquirer, history: new Map() };
+  for (const [stepName, step] of Object.entries(steps)) {
     const stepData = await step.isApplicable(context);
     if (stepData) {
       process.stdout.write('\n');
       await step.run(context, stepData);
+      context.history.set(stepName, []);
```

Comment:
I'd add it before `step.run` as when we will work on _telemetry_ part, I believe we will add data to step history in context of `step.run` invocation.

Also we may introduce `context.stepHistory` which will point added array directly (so will have to be overriden prior each `step.run` (and probably deleted prior `step.isApplicable`). So from context of a `step` we will not have to access it via `history.get(<this-step-name>)`

Reply:
Great call :+1: I think we will also have to return either whole context or `history` from the interactive run, but that will be needed when setting up the telemetry reporting, so I didn't add it for now.